Feat/lang dropdown sreya

tiramisuuuuuuu opened the pull request:
This code is for the language dropdown feature! 

A couple of side notes :(
- I have added a state variable to track the selected language and kept it as a context from the index page, so all children of the index page would be able to access the language variable. The dropdown itself accesses this state variable as a child of the index, not as a direct owner of the variable.

- I have hardcoded the width of the dropdown box in order to match the hardcoded rem of the text.

- I brought in the dividers, arrows, and symbols as svg files because they were not looking or sizing right as react-icons.

Could you please let me know if these will be okay towards the project or if I could make any improvements? :)

tiramisuuuuuuu committed:
Made a functioning dropdown

Created the functional aspects of the dropdown: when the v button is clicked, a table of options is created; the clicking of different languages is saved into a state variable; and the component that shows the selected language gets rerendered based on the state variable. Also, I saved the state variables into a context that hopefully allows other components to be able to reference the language selection : )

tiramisuuuuuuu committed:
Moved the LanguageContext to a higher page

In the previous commit, I created a context to store the "language" state variable. In this commit, I moved the context to the index page, so all the child components of the index will get access to the context: the currently selected language and a function to change the language (I put the function for the dropdown component to access TT)

tiramisuuuuuuu committed:
Created hover function and close when clicking out of div function

tiramisuuuuuuu committed:
Added some animation and styles

Made the dropdown rollout animation!

tiramisuuuuuuu committed:
Added icons and dividers to dropdown

tiramisuuuuuuu committed:
Added more styling details!

tiramisuuuuuuu committed:
Finalized dropdown!! (added original language names and fixed context values)

Made the language names to be written in the corresponding language. Made the returned context values of each option to match the names of json's (en, zh, ko, vi, hmn, and es) :)

JayJ104 commented on public/dropDown/arrowDown.svg:
I know you mentioned having sizing issues with react-icons but do you wanna elaborate on what issue it was again? We kinda prefer the icons and you can use font-size and rem units to resize it. If you haven't tried that, I'd recommend doing that but if it still doesn't work let me know.

tiramisuuuuuuu committed:
Added i18 features in dropdown

tiramisuuuuuuu committed:
Added i18n compatibility + Condensed dropdown css & code

I deleted repetitive code in dropDown.jsx using object mapping & created a i18n.config. I also left an example, compatible to my additions in dropDown.jsx, in index.jsx on how I am imagining i18n would work

tiramisuuuuuuu committed:
Replaced imgs with React Icons + Cleaned up dropDown css

JayJ104 committed:
Merge branch 'main' into feat/lang_dropdown-sreya

JayJ104 committed:
Merge branch 'feat/lang_dropdown-sreya' of https://github.com/include-davis/paul-homm into feat/lang_dropdown-sreya

JayJ104 committed:
language change simplified
